packages/sanity/src/structure/comments/src/components/list/CommentsListItemLayout.tsx

```diff
@@ -156,6 +150,20 @@ export function CommentsListItemLayout(props: CommentsListItemLayoutProps) {
   const hasChanges = useCommentHasChanged(value)
   const hasValue = useMemo(() => hasCommentMessageValue(value), [value])
 
+  // Filter out reactions that's been optimistically removed from the comment.
+  const reactions = useMemo(
+    () =>
+      (comment?.reactions?.filter((r) => r?._optimisticState !== 'removed') || EMPTY_ARRAY).filter(
+        (r) => {
+          // Filter out reactions that might have to incorrect format
+          return 'userId' in r && 'shortName' in r
```

Comment:
Curious in how we could get to this situation and if we should be adding a safeguard like this.

Reply:
It is unlikely that we that comments will be in the wrong format – but since the comment document can be mutated via e.g. the client, one might add a reaction item that is in the wrong shape.